tests/interval/test_initialization.py

```diff
@@ -191,14 +191,58 @@ class TestTypeGuessing(object):
     @mark.parametrize(
         ('number_range', 'type'),
         (
-            ((2, 3), int),
-            ([-6, 8], int),
-            (8.5, float),
-            ([Decimal(2), 9], int),
-            ([Decimal('0.5'), 9], float),
-            ([date(2000, 1, 1), inf], date),
-            (('a', 'e'), str),
+                ((2, 3), int),
+                ([-6, 8], int),
+                (8.5, float),
+                ([Decimal(2), 9], int),
+                ([Decimal('0.5'), 9], float),
+                ([date(2000, 1, 1), inf], date),
+                (('a', 'e'), str),
         )
     )
     def test_guesses_types(self, number_range, type):
         assert Interval(number_range).type == type
+
+
+class TestIntervalChanging(object):
+    @mark.parametrize(
+        ('constructor', 'number_range', 'bad_lower'),
+        (
+                (IntInterval, (1, 2), 3),
+                (IntInterval, [1, 2], 3),
+                (IntInterval, (1, 2), float('inf')),
+                (CharacterInterval, ('a', 'b'), 'c'),
+                (CharacterInterval, ('a', 'b'), 'd'),
+                (CharacterInterval, ('a', 'b'), inf),
+        )
+    )
+    def test_raises_exception_for_badly_lower_changing(
+            self,
+            constructor,
+            number_range,
+            bad_lower
+    ):
+        with raises(RangeBoundsException):
+            interval = constructor(number_range)
+            interval.lower = bad_lower
+
+    @mark.parametrize(
+        ('constructor', 'number_range', 'bad_upper'),
+        (
+                (IntInterval, (1, 2), 0),
```

Comment:
Use the indentation of 4 spaces not 8.

Reply:
The indentation has been revised to 8.

Reply:
There are still lines with indentation of 8 not 4

Reply:
OK, I will check again and change them.